api/api/src/main/scala/uk/ac/wellcome/platform/api/swagger/SwaggerScalaModelConverter.scala

```diff
@@ -0,0 +1,204 @@
+/* IMPORTANT:
+ *
+ * This file contains code for generating swagger docs from Scala case classes, and is mirrored from:
+ * https://github.com/swagger-akka-http/swagger-scala-module
+ *
+ * I could not get it working by depending on that library, as encountered the following runtime error:
+ * java.lang.NoSuchMethodError: com.fasterxml.jackson.module.scala.experimental.ScalaObjectMapper
+ * This appears to be due to differences in Jackson version needed for our version of Scala and that
```

Comment:
Why do we still have Jackson though? Maybe this has changed while I was away, but I thought that we were using CIrce for everything other than Finatra?

Reply:
It's needed for the generation of swagger JSON unfortunately :cry:

Reply:
I think it is because the library uses a version of Jackson which uses scala 2.13:

https://github.com/swagger-akka-http/swagger-scala-module/blob/ca1e802b856b86331ff3879cefe13c3f0a2f34de/build.sbt#L28

I am not entirely certain though, so if you want to investigate how to get it working without the copy/paste please feel free!

Reply:
I see, thanks for explaining. 
I'm not massively keen on the copy paste also because it's not incredibly intelligible to me what it does, but I don't think we should hold off merging because of this. If we can manage to ged rid of this later on then great, otherwise 🤷‍♀

Reply:
:+1: yep totally agree that would be better to avoid doing this, copy / pasting code which I don't really understand is not something I want to be in the habit of doing!